### PR TITLE
warn gradtape users  about tf.function support

### PR DESCRIPTION
**Pull request by NihalHarish**
#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smdebug/tensorflow/keras.py [hunk 1]
```diff
@@ -539,7 +539,8 @@ def _save_layer_input_and_outputs(self):
                 if self._is_collection_being_saved_for_step(CollectionKeys.LAYERS)
                 else set()
             )
-            self._save_tensor_to_file(export_name, tensor.numpy(), input_collection)
+            if hasattr(tensor, "numpy"):
```
Comment (vandanavk) [inline smdebug/tensorflow/keras.py:542]: do we have to warn the user if this cannot be saved?

File: tests/tensorflow2/test_grad_tape_tf_function.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
+            tf.keras.layers.Dense(128, activation="relu"),
+            tf.keras.layers.Dropout(0.2),
+            tf.keras.layers.Dense(10, activation="softmax"),
+        ]
+    )
+    return model
+
+
+def test_gradtape_tf_function(out_dir):
```
Comment (vandanavk) [inline tests/tensorflow2/test_grad_tape_tf_function.py:27]: do we have to update the quip doc or general docs about what is supported in this combination?

Comment (NihalHarish) [reply tests/tensorflow2/test_grad_tape_tf_function.py:27]: Yes. Or we can add this example to the examples repository.

File: smdebug/tensorflow/keras.py (hunk 1 above)
Comment (NihalHarish) [reply smdebug/tensorflow/keras.py:542]: Done

Comment (vandanavk) [reply smdebug/tensorflow/keras.py:542]: will this below log then be called for every tensor that fails?

Comment (NihalHarish) [reply smdebug/tensorflow/keras.py:542]: The function short circuits after printing the log once for every step.

**Review by vandanavk — APPROVED** [review]

**Pull request merged**